protobuf-lite/src/main/java/io/grpc/protobuf/lite/ProtoLiteUtils.java

```diff
@@ -107,23 +110,38 @@ public T parse(InputStream stream) {
             }
           }
         }
+        CodedInputStream cis;
+        if (stream instanceof ReadableBuffer) {
```

Comment:
Given that @nmittler is going to make this all better soon, lets choose an arbitrary size for the better path. 

64k anyone? Thats quite a lot of protobuf